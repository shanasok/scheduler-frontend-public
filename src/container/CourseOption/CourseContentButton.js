import React from 'react';
import { Link } from "react-router-dom";

export class CourseContentButton extends React.Component {
    render() {
        return <Link to={{ pathname: "/course",  state:{ course: this.props.course } }}>
            <button className="curvedButton">
                View Course Content
            </button>
        </Link>
    }
}