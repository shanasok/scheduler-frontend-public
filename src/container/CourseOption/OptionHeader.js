import React from 'react';

export class OptionHeader extends React.Component {

    render() {
        return <div className="optionHeader">
            {this.props.name}
        </div>
    }
}