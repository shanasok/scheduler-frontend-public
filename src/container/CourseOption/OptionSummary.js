import React from 'react';

export class OptionSummary extends React.Component {
    render() {
        return <div className="optionSummary">
            {this.props.summary}
        </div>
    }
}