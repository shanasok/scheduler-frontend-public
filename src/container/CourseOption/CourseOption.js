import React from 'react';

import {OptionHeader} from "./OptionHeader";
import {OptionSummary} from "./OptionSummary";
import {CourseContentButton} from "./CourseContentButton";
import {CourseStructuredData} from "../structuredData/CourseStructuredData";

export class CourseOption extends React.Component {

    render() {
        if (this.props.course)
            return <div className="courseOption">
                <OptionHeader name={this.props.course.name}/>
                <OptionSummary summary={this.props.course.summary}/>
                <CourseStructuredData course={this.props.course}/>
                <CourseContentButton course={this.props.course} key={`courseStructuredDataJSON-${this.props.course.id}`}/>
            </div>
        else
            return null;
    }
}