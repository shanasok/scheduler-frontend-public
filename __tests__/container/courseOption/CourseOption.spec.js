import React from 'react';
import {CourseOption} from '../../../src/container/courseOption/CourseOption';
import renderer from 'react-test-renderer';
import {BrowserRouter} from "react-router-dom";


it('renders correctly', () => {
    const course = {
        name: "java",
        summary: "java summary"
    }
    const tree = renderer
        .create(<BrowserRouter><CourseOption course={course}/></BrowserRouter>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders correctly when no course is passed in', () => {
    const tree = renderer
        .create(<BrowserRouter><CourseOption /></BrowserRouter>)
        .toJSON();
    expect(tree).toBeNull();
});