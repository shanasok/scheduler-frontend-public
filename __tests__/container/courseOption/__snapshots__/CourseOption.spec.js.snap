// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="courseOption"
>
  <div
    className="optionHeader"
  >
    java
  </div>
  <div
    className="optionSummary"
  >
    java summary
  </div>
  <script
    type="application/ld+json"
  >
    {"@context":"http://schema.org","@type":"Course","name":"java","description":"java summary"}
  </script>
  <a
    href="/course"
    onClick={[Function]}
  >
    <button
      className="courseContentButton"
    >
      View Course Content
    </button>
  </a>
</div>
`;
